internal/provider/file/image.go

```diff
@@ -31,10 +30,15 @@ func (c *Client) listFileImage() []model.Image {
 			c.logger.Error().Err(err).Msgf("Unable to decode into struct %s", file)
 			continue
 		}
+
 		for _, item := range items {
+			// Set default WatchRepo
+			if !item.WatchRepoWasSet() {
```

Comment:
Related to my comment about `WatchRepo` you would just need:

```suggestion
			if c.imageDefaults.WatchRepo != nil {
```